Pod/Classes/UISideMenuNavigationController.swift

```diff
@@ -147,17 +147,52 @@ open class UISideMenuNavigationController: UINavigationController {
         if SideMenuManager.menuAllowPopIfPossible {
             for subViewController in navigationController.viewControllers {
                 if type(of: subViewController) == type(of: viewController) {
+                    if SideMenuManager.menuTabMode {
+                        navigationController.popToViewController(subViewController, animated: false)
+                        CATransaction.commit()
+                        return
+                    }
                     navigationController.popToViewController(subViewController, animated: animated)
                     CATransaction.commit()
                     return
                 }
             }
         }
         
+        if SideMenuManager.menuPreserveViewOnPush {
+            var viewControllers = navigationController.viewControllers
+            let filtered = viewControllers.filter{$0.restorationIdentifier == viewController.restorationIdentifier}
+            if let preservedView = filtered.first {
+                viewControllers = viewControllers.filter() { $0 !== preservedView }
+                viewControllers.append(preservedView)
+                if SideMenuManager.menuTabMode {
+                    navigationController.setViewControllers(viewControllers, animated: false)
```

Comment:
Yes, good point - the expectation is for the user to choose the view controller from the sidemenu, rather than using the back arrow. However, when a back arrow is involved, the SideMenu will not be selectable anyway - the default behavior is to restore the last view controller, typically the root controller, and not to display the SideMenu. 

The flow: from the root controller, the user will be able to select their SideMenu view controller, and enjoy a preserved view controller. If a back arrow button is involved, pressing it will take the user back to the root controller. If it is not involved, the user will be able to present the SideMenu without problem, and choose their new respective view controller.